plugins/modules/zos_backup_restore.py

```diff
@@ -439,6 +439,7 @@ def main():
                 sms_management_class=sms_management_class,
                 tmp_hlq=tmp_hlq,
             )
+            result["backup_name"] = backup_name
```

Comment:
Backup name is only being returned when operation is `backup`, that is wrong, it should be returned in all successful cases as mentioned in the issue. Took this output from the Jenkins tests, is a result when operation is restore. Please add backup name in all successful cases.

```
(0, b'\n{"changed": true, "message": "", "backup_name": "", "invocation": {"module_args": {"operation": "restore", "backup_name": "ANSIBLE.PO.T1854032.CK", "hlq": "NR8XJ", "overwrite": false, "full_volume": false, "recover": false, "data_sets": null, "space": null, "space_type": null, "volume": null, "temp_volume": null, "sms_storage_class": null, "sms_management_class": null, "tmp_hlq": null}}}
```